### PR TITLE
bitfield dist logging cleanup and enhancements

### PR DESCRIPTION
**Pull request by drahnr**

### CODE REVIEW

File: node/network/bitfield-distribution/src/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+				prometheus::Counter::new(
+					"polkadot_parachain_received_availabilty_bitfields_total",
+					"Number of valid availability bitfields received from other peers.",
+				)?,
+				registry,
+			)?,
+			active_leaves_update: prometheus::register(
+				prometheus::Histogram::with_opts(prometheus::HistogramOpts::new(
+					"polkadot_parachain_bitfield_distribution_active_leaves_update",
+					"Time spent within `bitfield_distribution::active_leaves_update`",
```
[the remark this answers is not in the document]
Comment (drahnr) [reply node/network/bitfield-distribution/src/metrics.rs:87]: They were just moved out, so I did not change these lines. I think so far the default buckets were ok?

File: node/network/bitfield-distribution/src/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
```
Comment (sandreim) [inline node/network/bitfield-distribution/src/metrics.rs:1]:
```suggestion
// Copyright 2022 Parity Technologies (UK) Ltd.
```

File: node/network/bitfield-distribution/src/metrics.rs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		self.0.as_ref().map(|metrics| metrics.handle_network_msg.start_timer())
+	}
+}
+
+impl MetricsTrait for Metrics {
+	fn try_register(registry: &prometheus::Registry) -> Result<Self, prometheus::PrometheusError> {
+		let metrics = MetricsInner {
+			gossipped_own_availability_bitfields: prometheus::register(
+				prometheus::Counter::new(
+					"polkadot_parachain_gossipped_own_availabilty_bitfields_total",
```
Comment (sandreim) [inline node/network/bitfield-distribution/src/metrics.rs:72]: Match naming with `polkadot_parachain_received_availabilty_bitfields_total`
```suggestion
					"polkadot_parachain_sent_own_availabilty_bitfields_total",
```

File: node/network/bitfield-distribution/src/metrics.rs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	/// Provide a timer for `handle_network_msg` which observes on drop.
+	pub(crate) fn time_handle_network_msg(&self) -> Option<prometheus::prometheus::HistogramTimer> {
+		self.0.as_ref().map(|metrics| metrics.handle_network_msg.start_timer())
+	}
+}
+
+impl MetricsTrait for Metrics {
+	fn try_register(registry: &prometheus::Registry) -> Result<Self, prometheus::PrometheusError> {
+		let metrics = MetricsInner {
+			gossipped_own_availability_bitfields: prometheus::register(
```
Comment (sandreim) [inline node/network/bitfield-distribution/src/metrics.rs:70]:
```suggestion
			sent_own_availability_bitfields: prometheus::register(
```

**Review by sandreim — APPROVED** [review]

File: node/network/bitfield-distribution/src/metrics.rs (hunk 2 above)
Comment (drahnr) [reply node/network/bitfield-distribution/src/metrics.rs:1]: It's copied, so the license is correct as is

**Comment by drahnr** [comment]
bot merge

**Comment by drahnr** [comment]
bot merge

**Comment by sandreim** [comment]
bot merge

**Pull request merged**